src/unknown.rs

```diff
@@ -0,0 +1,50 @@
+use std::fs::File;
+use std::io;
+use std::path::Path;
+
+static ERROR_MESSAGE: &str = "Unknown Architecture";
+
+#[derive(Debug)]
+pub struct Handle;
+
+impl Eq for Handle {}
+
+impl PartialEq for Handle {
+    fn eq(&self, _other: &Handle) -> bool {
+        false
```

Comment:
I think you should put `unreachable!()` here. (See comment below about how it's impossible to get a `Handle` in the first place.)